### ISSUE TITLE
dayplans: filemaker shows blank panel over UI after inserting an image (error type 1)

### ISSUE THREAD

**Issue by urapadmin** [urapadmin/kiosk#1249]
iOS 14.7.2

I can do this now too. And here is proof that Filemaker  is still alive behind the grey panel. 

![9C3A0810-5531-471D-8868-B0596559DE7C](https://user-images.githubusercontent.com/38838314/140938213-517e7c6a-bb77-4dfe-9a3c-8bced563d8ff.png)

_Originally posted by @urapadmin in https://github.com/urapadmin/kiosk/issues/1246#issuecomment-964181051_

**Comment by urapadmin** [comment]
The solution to this is delaying the file-identifier-cache operations and run them asynchronously. That will be a particular challenge when it comes to locus relations and in general a real mess. 

There is also still the option on the table to delay it even further until the file identifier cache is actually needed OR the app is being closed (the latter being problematic). That would have the advantage that I would get rid of almost all the flickering.

**Comment by lbestock** [comment]
Wait, did your new iPad arrive early?
How do locus relation images work in the opposite direction and is that implicated? I mean, I know that they appear (as they should) when the opposite location is created. It seems to me far the most usual case where the same photograph appears in two places.

**Comment by urapadmin** [comment]
yes to both. 
All the image containers need to be rewritten, That will take days and every single one will need a new thorough systematic test.
So I make them all new tickets.

**Comment by urapadmin** [comment]
I won't be the only one who will hate this. This here is about dayplans:

### recording 11.6
- [ ] complete image container test in simple excavating. see in wiki: [https://wiki.arch-kiosk.brown.edu/urapdev/doku.php?id=how_to_test_an_image_container&s[]=image&s[]=container](https://wiki.arch-kiosk.brown.edu/urapdev/doku.php?id=how_to_test_an_image_container&s[]=image&s[]=container). 
- [ ] complete image container test in full excavating

**Comment by luizaogs** [comment]
Testing this and most of it seems fine. But when I tried to add a dayplan from the gallery I got this: 

![IMG_1B77D5F3E07C-1](https://user-images.githubusercontent.com/67970402/141323074-441736a9-a728-4327-ad3d-29c16b9c9222.jpeg)

So here I am, informing the developer.

**Comment by luizaogs** [comment]
I think that's all I have!

**Comment by urapadmin** [comment]
Following a lecture on Kant's ethics aside that turned into a discussion with students saying sentences starting with "They tought us in school that ... ", so I am glad to respond to something worth the while:

> 
> 
> Hmm, a potential second problem. I deleted a dayplan to check if it would no longer appear under the unit identifier in the gallery, which indeed was the case and seemed proper, but I can no longer find it under the image description so it seems like the image itself has been deleted from the gallery?
> 
> Yeah alright, it's definitely not in there.
> 
> (To myself, so I remember what I am talking about: unit is Blah in Playground and image description was Test 2)


That is correct. That description is an attribute of  the relation between the image and the context. It disappears indeed when that relationship is deleted. In the gallery you have a button that shows you details about the image and if you click on that you see those descriptions always bound to the context. 

So that is correct behaviour.

**Comment by urapadmin** [comment]
> 
> 
> I think that's all I have! The rest is fine.

That sounds promising. The gallery issue is just something I forgot to adapt. That is why we need to test this stoically. I will forget such things in other containers again since I have to adapt many code passages for every single container again and again. The testing hell mirrors the coding hell.

**Comment by luizaogs** [comment]
> Following a lecture on Kant's ethics aside that turned into a discussion with students saying sentences starting with "They tought us in school that ... ", so I am glad to respond to something worth the while:
> 
> > Hmm, a potential second problem. I deleted a dayplan to check if it would no longer appear under the unit identifier in the gallery, which indeed was the case and seemed proper, but I can no longer find it under the image description so it seems like the image itself has been deleted from the gallery?
> > Yeah alright, it's definitely not in there.
> > (To myself, so I remember what I am talking about: unit is Blah in Playground and image description was Test 2)
> 
> That is correct. That description is an attribute of the relation between the image and the context. It disappears indeed when that relationship is deleted. In the gallery you have a button that shows you details about the image and if you click on that you see those descriptions always bound to the context.
> 
> So that is correct behaviour.

So then how do I check if the image is still in the gallery? Or do I not have to? I thought I did because of the wiki.

**Comment by urapadmin** [comment]
> 
> 
> > Following a lecture on Kant's ethics aside that turned into a discussion with students saying sentences starting with "They tought us in school that ... ", so I am glad to respond to something worth the while:
> > > Hmm, a potential second problem. I deleted a dayplan to check if it would no longer appear under the unit identifier in the gallery, which indeed was the case and seemed proper, but I can no longer find it under the image description so it seems like the image itself has been deleted from the gallery?
> > > Yeah alright, it's definitely not in there.
> > > (To myself, so I remember what I am talking about: unit is Blah in Playground and image description was Test 2)
> > 
> > 
> > That is correct. That description is an attribute of the relation between the image and the context. It disappears indeed when that relationship is deleted. In the gallery you have a button that shows you details about the image and if you click on that you see those descriptions always bound to the context.
> > So that is correct behaviour.
> 
> So then how do I check if the image is still in the gallery? Or do I not have to? I thought I did because of the wiki.

There is only one way, unfortunately. You have to prepare an image that you can identify in the gallery and that you have not used for anything else. New images are usually added to the end of the gallery. That might help you. This is in fact something that would be easier to test in a system with less data. 💡 Oh, but now that we have file picking you could make a workstation with no images! And then the gallery should have nothing but the image you are testing with.

**Comment by luizaogs** [comment]
Goddammit. I downloaded a database with all dummy images to test the gallery thing and now am GETTING 2bs AGAIN. What? I assume this should not be a side effect of telling all contextuals to be dummies, so this is not good.

Three dayplans, three 2bs. Bloody hell.

**Comment by luizaogs** [comment]
> > > Following a lecture on Kant's ethics aside that turned into a discussion with students saying sentences starting with "They tought us in school that ... ", so I am glad to respond to something worth the while:
> > > > Hmm, a potential second problem. I deleted a dayplan to check if it would no longer appear under the unit identifier in the gallery, which indeed was the case and seemed proper, but I can no longer find it under the image description so it seems like the image itself has been deleted from the gallery?
> > > > Yeah alright, it's definitely not in there.
> > > > (To myself, so I remember what I am talking about: unit is Blah in Playground and image description was Test 2)
> > > 
> > > 
> > > That is correct. That description is an attribute of the relation between the image and the context. It disappears indeed when that relationship is deleted. In the gallery you have a button that shows you details about the image and if you click on that you see those descriptions always bound to the context.
> > > So that is correct behaviour.
> > 
> > 
> > So then how do I check if the image is still in the gallery? Or do I not have to? I thought I did because of the wiki.
> 
> There is only one way, unfortunately. You have to prepare an image that you can identify in the gallery and that you have not used for anything else. New images are usually added to the end of the gallery. That might help you. This is in fact something that would be easier to test in a system with less data. 💡 Oh, but now that we have file picking you could make a workstation with no images! And then the gallery should have nothing but the image you are testing with.

Beyond the issue above, I did try to make sure the image remained in the gallery after I deleted the dayplan and it does not. I only had two dayplans work (the others are the blank image containers courtesy of 2b) and saw both in the gallery. Deleted one of them from the list of dayplans, now I only get one image in the gallery -- easy to see since everything else is a dummy. So it seems like this is currently also an actual issue, unless I misunderstood your instructions.

**Comment by urapadmin** [comment]
I stop the quoting, it gets a bit overindented. No, you understood me fairly well and what you report is a strange thing, indeed.

**Comment by urapadmin** [comment]
> 
> 
> Goddammit. I downloaded a database with all dummy images to test the gallery thing and now am GETTING 2bs AGAIN. What? I assume this should not be a side effect of telling all contextuals to be dummies, so this is not good.
> 
> Three dayplans, three 2bs. Bloody hell. The fifth one worked, but clearly the issue is back.

your ipad must be kaputt. Please give me your exact iOS and filemaker versions again. And are you testing simple excavation or full? And you have not checked anything in the Guinea Pig section (didn't I even hide that in this version?!?)

**Comment by urapadmin** [comment]
Oh and please, next time you get another 2b again, please send me the test log right after the delayed flicker.

**Comment by luizaogs** [comment]
iOS 15.1, FM 19.3.1.23. Simple excavating, nothing in the Guinea Pig section (it is hidden).

**Comment by luizaogs** [comment]
add_dayplan_to_cache_deferred: single image with param OnObjectModify, $$cache_done=
add_dayplan_to_cache_deferred: image missing: 0/1
is_cache_update_needed: image is there in OnObjectModify but record is not-> DONT process cache
add_dayplan_to_cache_deferred: doing nothing.
image_validate: field img
add_dayplan_to_cache_deferred: single image with param , $$cache_done=0
add_dayplan_to_cache_deferred: image missing: 0/0
is_cache_update_needed: OnValidate but NOT CACHE_DONE-> process cache
add_dayplan_to_cache_deferred: cache update needed.
add_dayplan_to_cache_deferred: cache update timed.
activate_temp_window: opened layout file_identifier_cache in NEW window  temp_window on iOS
execute_deferred_cache_operation: deleted cache entries.
closed temp window  temp_window.

**Comment by urapadmin** [comment]
totally normal. I am clueless.

**Comment by luizaogs** [comment]
That makes two of us. It worked perfectly yesterday...

**Comment by urapadmin** [comment]
Wouldn't surprise me if you restarted it and then have another 10 dayplans free until it begins happening again.

**Comment by urapadmin** [comment]
Laurel will try to find an ipad for you to switch. That one might need to be sent to Germany.

**Comment by urapadmin** [comment]
>Beyond the issue above, I did try to make sure the image remained in the gallery after I deleted the dayplan and it does not. I only had two dayplans work (the others are the blank image containers courtesy of 2b) and saw both in the gallery. Deleted one of them from the list of dayplans, now I only get one image in the gallery -- easy to see since everything else is a dummy. 

I cannot reproduce that, either. I downloaded your Luiza-isms workstation, made three dayplans, deleted the middle one and while it stayed in the gallery it was disconnected from the context. 
- Can you reproduce this reliably? 
- If so, please be quite precise about how you delete (I suppose you were using the menu on the image container?), what you delete (the first dayplan in the list, the second, the last ...).
- I also wonder if the 2b behavior interfered in some way or the other with this

**Comment by urapadmin** [comment]
![image](https://user-images.githubusercontent.com/38838314/141472597-cc24dfb5-4a49-49f2-9727-9b73a699b319.png)

**Comment by luizaogs** [comment]
Ha no, I was deleting the actual dayplan (the red trash can icon). But I just tried deleting another with the delete button in the actions popup and the result seems to have been the same. These were neither first nor last in the list, they were in the middle.

**Comment by luizaogs** [comment]
WAIT. I'm stupid. LA was in the search field. Ignore all of this (it's early in the morning).

SO it's all fine (other than the 2b fiasco). Just the adding from the gallery message above. Sorry!

**Comment by urapadmin** [comment]
![banghead](https://user-images.githubusercontent.com/38838314/141473515-42362abd-c8c1-4416-acc3-56e458e8b29b.gif)

**Comment by urapadmin** [comment]
also deleting the dayplan should have the same result: It must not delete the image. In fact there is no place in the ui where you could possibly delete an image again!

**Comment by luizaogs** [comment]
It didn't. Finding 2bs again must have thrown me off and I didn't realize I had the unit identifier in the search box of the image gallery. The images are there, so you can ignore all of this nonsense.

**Comment by urapadmin** [comment]
as for the 2b:
![banghead](https://user-images.githubusercontent.com/38838314/141474691-e332e876-6bc5-4c3c-be50-2b2ab589fb15.gif)

**Comment by urapadmin** [comment]
> 
> 
> It didn't. Finding 2bs again must have thrown me off and I didn't realize I had the unit identifier in the search box of the image gallery. The images are there, so you can ignore all of this nonsense.

okay, phew, that would have been an entirely new oddity.

**Comment by luizaogs** [comment]
They may be out of fashion but um VERY accurate.

**Comment by urapadmin** [comment]
Normally things are pretty well isolated from one another on iOS, but I ask it anyways:

Can you think of anything you are using on that ipad or have installed that might not be mainstream? Some special app for some totally esoteric research involving the camera (analyzing aura photos of mummies comes to mind). I don't know. I just try to find out why in the world it hits you only (and no, it has nothing to with you personally or your own aura, I am sure!).

**Comment by urapadmin** [comment]
### recording v11.6.2
- [ ] inserting from the gallery should work properly now

**Comment by luizaogs** [comment]
> Normally things are pretty well isolated from one another on iOS, but I ask it anyways:
> 
> Can you think of anything you are using on that ipad or have installed that might not be mainstream? Some special app for some totally esoteric research involving the camera (analyzing aura photos of mummies comes to mind). I don't know. I just try to find out why in the world it hits you only (and no, it has nothing to with you personally or your own aura, I am sure!).

Nope. Sadly nothing comes to mind.

**Comment by luizaogs** [comment]
Ok well the gallery works now but I also now get 2as instead of 2bs on this iPad, so cool. Will check if things are working on the new one in a bit…

**Comment by urapadmin** [comment]
you get 2as only when using the gallery or in general? The 2as could really be a bug, we must be careful not to attribute future issues to the strange ipad. Perhaps it is best, indeed, if you pause until you have the brand new one.

Oh, that reminds me: Should you get the new one and it is still on iOs 15.0.2 (you know, the os version we thought was responsible for the 2as), please try first if you can reproduce the 2as using that version before you update the OS.

**Comment by luizaogs** [comment]
So yeah I failed on the not updating front because I didn't think of trying out 15.0.2 myself and only saw your message after the (somewhat automated) update had gone through. Apologies.

Good news, though, is that I seem to not yet have broken the new iPad :) Everything is working with dayplans now, including the gallery. So I think we can close this one, no?

**Comment by luizaogs** [comment]
(I like the new PVD logo, by the way)

**Comment by urapadmin** [comment]
I am reluctant to close anything these days. But yes. let's close it and if the 2b or 2a resurfaces, we open a new ticket for it.

**Comment by urapadmin** [comment]
### test 11.6.14
After the long phase on site maps and site notes I had to reopen this. It needs another round of tests, too.

**Comment by lbestock** [comment]
Well, and I feel we should add @mcornacchia25 to this one, too. Because it is particularly on testing image containers that we will need your help in the near future, Meghan! And so here you can see (though it is baroque, so presumably not immediately understand) a re-opened image container ticket needing the full treatment detailed [in the wiki](https://wiki.arch-kiosk.brown.edu/urapdev/doku.php?id=how_to_test_an_image_container).

**Comment by luizaogs** [comment]
It's all working for me.

**Comment by urapadmin** [comment]
that's something. But at this point we can close these image container tickets only after they have been tested on more than one device. On mine it is fine, too (but I lack patience, currently, to force it).

**Comment by urapadmin** [comment]
for Meghan I add here that when testing anything about "dayplans" we have to test two different things: dayplans in simple excavation mode and dayplans in full excavation mode. 

First of all, this is only relavant to units of type "excavation/locus recording". So to test this you need one of those first.

Then in the configuration you can switch between the two modes as admin by going to "visibility/use_simple_excavating". Enter either "yup" or "nope". You will see that depending on the setting the excavation or "digging" form will have more or less tabs.

Once you have entered "yup" here, a feature for testers is unlocked: On the start page, tab "registrar and specialists" you'll find a new checkbox in the lower right corner. This allows you to switch between simple and full excavation temporarily without having to enter the configuration all the time.

**Comment by lbestock** [comment]
Alright, testing team. With these guys we are not satisfied with one tester. Let's have a policy on all these image container tests that all three of us, @luizaogs, @mcornacchia25, and @lbestock, must test and find no problems. Third tester to find no problems closes the ticket. Luiza, you've done this one. Meghan, you and I need to do it. And I'm going to add you to a few other tickets.

**Comment by mcornacchia25** [comment]
Going through the instructions for the image container test, I'm a bit confused. What are / how do I find the primary image container and the gallery?

**Comment by luizaogs** [comment]
Hi Meghan :)

The image gallery is accessible on the top right of the start page of the database, after you log in (you should see a button there). For dayplans, the primary image containers are accessible on the unit page. So if you pick unit LA in Providence, for instance, and click on the dayplan/narrative tab you can add new dayplans by clicking the plus button. That white container that appears once you add a new dayplan is the primary image container (you add the image there by clicking on it and choosing Camera). The popup, on the other hand (which you will also see in the instructions for the test), is accessible if you click on the small button on the top right of the primary image container (I'm including a screenshot below). From there you can also add new images by clicking on the white part of the container, and that's also where you find the "insert from gallery" button.

![IMG_8946ACC1D2A6-1](https://user-images.githubusercontent.com/67970402/147148438-289bdc14-af78-49ce-ac60-ccf0f4d25b55.jpeg)

(There are primary image containers and popups for the different types of tests we're doing, but they mostly work the same across the board.)

Laurel probably already told you this but there are a couple of in-progress user manuals on the start page of the wiki. Hope this helps!

**Comment by mcornacchia25** [comment]
> Once you have entered "yup" here, a feature for testers is unlocked: On the start page, tab "registrar and specialists" you'll find a new checkbox in the lower right corner. This allows you to switch between simple and full excavation temporarily without having to enter the configuration all the time.

I checked with admin and this setting was turned on, but I'm not seeing this checkbox. Should I just manually turn it on and off in testing or is there some way to get it to show up that I'm missing?


![IMG_7F7C7900889E-1](https://user-images.githubusercontent.com/96021337/147155758-36367956-8a61-459d-8c30-36e6af94d406.jpeg)

**Comment by mcornacchia25** [comment]
I manually switched and followed the guide. Everything seemed to work as expected.

**Comment by urapadmin** [comment]
I am sorry, I go this wrong: "use_simple_excavating " has to be "nope" (sorry for these yups and nopes, they started as a joke and now I can't get rid of them anymore) and you have to have registrar status (which you have) and then the checkbox should show:
![image](https://user-images.githubusercontent.com/38838314/147271524-bd53e9d8-222e-4f35-b293-6a83d7530ab2.png)

**Comment by urapadmin** [comment]
I use this opportunity to tell you about test-mode: If you switch admin/test_mode on on (yup again), you get access to a few additional buttons that make you life as a tester a little bit easier: 
- a gallery button on each form to give you easy access to check things in the gallery.
- If things go wrong with an image container you also have the "test log" button that shows a log that can give me hints when analyzing the issue. But I will ask you to provide it if I think it might be useful.
- a "file identifier cache" button on the start page that can sometimes he helpful to look up the file identifier information for an image from a somewhat different perspective.

**Comment by urapadmin** [comment]
so, can we close this ?
(but beware of #1251 )

**Comment by luizaogs** [comment]
It's all still working for me here!

**Comment by urapadmin** [comment]
please close this only once #1251 is okay, too. They should behave the same way.

**Comment by urapadmin** [comment]
### @test 11.6.17
Sorry we have to test this again because of a bug I found. There should be exactly ONE flicker and all should work, of course.

**Comment by mcornacchia25** [comment]
Everything seems to be working as expected for me, but it looks like 2 quick flashes when adding an image.

**Comment by mcornacchia25** [comment]
Actually, I've realized I'm still on 11.6.16. How do I change the version I'm downloading from the kiosk?

**Comment by urapadmin** [comment]
by going to kiosk and preparing your workstation again :o) I just reset them all.
It makes sense to subscribe to this page in the wiki: [https://wiki.arch-kiosk.brown.edu/urapdev/doku.php?id=the_release_history_is_here#section11616](https://wiki.arch-kiosk.brown.edu/urapdev/doku.php?id=the_release_history_is_here#section11616). As soon as I roll out a new release I add it on that page. Before it does not appear on this list, you can't test a ticket that needs a higher version.

**Comment by mcornacchia25** [comment]
Good to know, thanks! Testing day plans in the new version, adding a new image works as expected with one flash, but adding from the gallery does not work. The images are not found under the new identifiers in the gallery, and they don't show up within the day plan until you click on another field. Changing the description does not fix the issue with images from the gallery and the description does not update even for the otherwise working images.

**Comment by luizaogs** [comment]
Hm, weird -- I am not done testing this yet but am not getting any of the issues mentioned by Meghan above.

**Comment by luizaogs** [comment]
Yeah, this is all working properly for me.

**Comment by mcornacchia25** [comment]
I think I know what happened. I tried to add from gallery again and it worked. But then I tried to click out of something too fast and it gave a "Stop this script" message. I clicked no, but after that, I had the same issues as above. Does interrupting the script even if you click no cause issues?

**Comment by luizaogs** [comment]
Ah yes, that problem is going to be dealt with in #1264.

**Comment by mcornacchia25** [comment]
Going to another tab and back fixes it.

**Comment by urapadmin** [comment]
@mcornacchia25 so without the script cancellation everything works for you? (by the way the images you inserted when cancelling can cause all kinds of other issues later on, so you should get rid of them).

**Comment by mcornacchia25** [comment]
Otherwise it does seem to work. The only things I tried to insert which had issues were already in the gallery. Should I just delete the dayplans or is there a way to delete them from the gallery?

**Comment by urapadmin** [comment]
you should definitely first delete them from the gallery. If you are admin you can go to the gallery and use the popup button on an image. It leads you to the identifiers and description. There should be a delete button now that drops an image for good.

Of course another particularly safe way to start testing from scratch is just downloading your workstation again. In fact I would prefer that whenever you want to make sure that you repeat a test without interference from the remains of earlier tests.

**Comment by mcornacchia25** [comment]
Fresh download sounds good

**Comment by urapadmin** [comment]
so are we done with this one? @lbestock?

**Comment by lbestock** [comment]
We are! I got a few 2s on the way, but no 1s. I begin to forget what they were like, the 1s.
